js/components/channel_messages.jsx

```diff
@@ -36,6 +36,7 @@ var ChannelMessages = React.createClass({
   },
 
   componentWillUnmount: function() {
+    MessageStore.removeListener('change', this._messageStoreChange);
```

Comment:
Sorry I misread your comment. I'm a little hazy on Flux still, so I don't exactly why it's happening. I know it has something to do with the way we're loading messages into the store. My initial assumption is that when the component was unmounted then mounted again the store would just reinitialize and fetch all the messages from Firebase but that wasn't happening...